service.onmute/service.py

```diff
@@ -0,0 +1,203 @@
+        self.userCC = self.getCC()
+        self.Player = Player()
+        self.Monitor = Monitor()
+        self.autoSub = False
+        self.Player.subFound = False
+        self.searchSub = self.Monitor.searchSub
+        self.start()
+
+        
+    def searchSubtitle(self):
+        log("searchSubtitle")
+        self.Player.subFound = True
+        xbmc.executebuiltin("ActivateWindow(SubtitleSearch)")
+        
+        
+    def isSubtitle(self):
+        state = xbmc.getCondVisibility('VideoPlayer.SubtitlesEnabled') == 1
+        log("isSubtitle = " + str(state))
+        return state
+         
+         
+    def setSubtitle(self, state):
+        log("setSubtitle = " + str(state))
+        self.Player.showSubtitles(state)
+         
+         
+    def hasSubtitle(self):
+        state = xbmc.getCondVisibility('VideoPlayer.HasSubtitles') == 1
+        log("hasSubtitle = " + str(state))
+        return state
+
+        
+    def isMute(self):
+        state = xbmc.getCondVisibility('Player.Muted') == 1
+        log("isMute = " + str(state))
+        return state
+
+
+    def getCC(self):
+        # save user parsecaptions settings prior to change
+        state = False
+        json_query = '{"jsonrpc":"2.0","method":"Settings.GetSettingValue","params":{"setting":"subtitles.parsecaptions"}, "id": 1}'
+        json_responce = loadJson(sendJSON(json_query))
+        if 'result' in json_responce and 'value' in json_responce:
+            state = json_responce['result']['value']
+        log("getCC = " + str(state))
+        return state
+        
+    
+    def setCC(self, state):
+        # change parsecaptions setting
+        log("setCC = " + str(state))
+        json_query = '{"jsonrpc":"2.0","method":"Settings.SetSettingValue","params":{"setting":"subtitles.parsecaptions","value":%s}, "id": 1}'%str(state).lower()
+        sendJSON(json_query)
+        
+    
+    def start(self):
+        # enable closed caption if user enables.
+        if self.Monitor.enableCC == True and self.getCC() == False:
+            self.setCC(True)
+            
+        while not self.Monitor.abortRequested():
+            if self.Player.isPlayingVideo() == True:
+                log("autoSub = " + str(self.autoSub))
+                log("subFound = " + str(self.Player.subFound))
+                
+                if xbmcgui.Window(10000).getProperty("PseudoTVRunning") == "True":
+                    self.searchSub = self.Monitor.searchSub
+                    self.Monitor.searchSub = False
+                else:
+                    self.Monitor.searchSub = self.searchSub
+                
+                '''
+                    missing subs return false positive isSubtitle = True,
+                    compare with hasSubtitle to detect true status.
+                '''
+                # check if user has subs already enabled.
+                if (self.isSubtitle() + self.hasSubtitle()) > 1 and self.isMute() == True and self.autoSub == False:
```

Comment:
No problem with styling at all, if it's up to me it can go in like this. But it's pretty weird if you mix compare styles in the same line, really screams to cause problems somewhen :)